File: src/main/resources/output/IFC4X3_java/Types/IfcLayeredItem.java
// SELECT TYPE IfcLayeredItem
public class IfcLayeredItem {
    public Object value;

    public IfcLayeredItem(Object value) {
        if (!(value instanceof IfcRepresentation || value instanceof IfcRepresentationItem)) {
            throw new IllegalArgumentException("Value must be one of: IfcRepresentation, IfcRepresentationItem");
        }
        this.value = value;
    }

    public <T> T getValueAs(Class<T> clazz) {
        return clazz.cast(value);
    }
}

File: src/main/resources/output/IFC4X3_js/Types/IfcLayeredItem.js
// SELECT TYPE IfcLayeredItem
// Options:
// - IfcRepresentation
// - IfcRepresentationItem
class IfcLayeredItem {
    constructor(value) {
        if (!(value instanceof IfcRepresentation || value instanceof IfcRepresentationItem)) {
            throw new Error("Value must be one of: IfcRepresentation, IfcRepresentationItem");
        }
        this.value = value;
    }

    getValueAs(TypeClass) {
        if (this.value instanceof TypeClass) {
            return this.value;
        } else {
            throw new Error("Stored value is not of the requested type");
        }
    }
}
